### PR TITLE
refactor: standardize id regex to {17,19}

### PR DESCRIPTION
**Pull request by onerandomusername**
The oldest ID is 17 characters, and 20 digit IDs won't be around for decades (if ever)

## Summary

Gets rid of ID regexes checking for 15 character IDs, and switches to a max of 19 characters because that's the maximum ID length for the foreseeable future.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Note: there are plans to eventually standardize all of these regexes and possibly compile them before usage, but that's out of the scope of this pr which just fixes up the regexes to be a bit more standardised.

**Comment by shiftinv** [comment]
There seems to be a mix of `17,19` (changelog, code) and `17,20` (PR, commit name) here, which want did you want to go with? `17,20` would be preferable since that's also what the client currently uses, even if it won't make a difference for the next 68 years.

**Comment by onerandomusername** [comment]
Meant to go with 17,19, commit title disagreed

**Comment by onerandomusername** [comment]
> 17,20 would be preferable since that's also what the client currently uses

This doesn't really make much of a difference, as the client will use 20 but still result in a snowflake that doesn't exist (as Discord resolves the object server side).

File: changelog/651.misc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Update and standardise all internal Snowflake regexes to be between 17 and 19 characters (inclusive).
```
Comment (shiftinv) [inline changelog/651.misc.rst:1]:
```suggestion
Update and standardise all internal Snowflake regexes to match between 17 and 19 characters (inclusive).
```

File: disnake/webhook/async_.py [hunk 2]
```diff
@@ -1193,7 +1193,7 @@ def from_url(
             A partial webhook is just a webhook object with an ID and a token.
         """
         m = re.search(
-            r"discord(?:app)?.com/api/webhooks/(?P<id>[0-9]{17,20})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})",
+            r"discord(?:app)?.com/api/webhooks/(?P<id>[0-9]{17,19})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})",
```
Comment (shiftinv) [inline disnake/webhook/async_.py:1196]: Do we know what the token length depends on? I just looked at a few tokens (including one that's more than 3 years old), they all had 68 chars which is already the upper limit for this regex :thinking:

**Review by shiftinv — COMMENTED** [review]
Looks good!
There are two more occurrences of `17,22` in `ext.commands.help` which would need to be updated as well.

File: disnake/webhook/async_.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/webhook/async_.py:1196]: I don't, but I don't think that is documented.

**Review by shiftinv — APPROVED** [review]

**Pull request merged**